### PR TITLE
feat(aio): code block styles with header

### PR DESCRIPTION
**Pull request by sjtrimble**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")
```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: aio/src/app/embedded/code/code-example.component.ts [hunk 1]
```diff
@@ -16,7 +16,10 @@ import { Component, ElementRef, OnInit } from '@angular/core';
   selector: 'code-example',
   template: `
     <header *ngIf="title">{{title}}</header>
-    <aio-code [code]="code" [language]="language" [linenums]="linenums"></aio-code>
+    <aio-code *ngIf="title" class="headed-code" [code]="code" [language]="language" [linenums]="linenums"></aio-code>
+
+    <aio-code *ngIf="!title" class="simple-code" [code]="code" [language]="language" [linenums]="linenums">
+    </aio-code>
```
Comment (petebacondarwin) [inline aio/src/app/embedded/code/code-example.component.ts]: we can do this more simply with a single element:

```html
<aio-code [class.headed-code]="title" [class.simple-code]="!title" [code]="code" [language]="language" [linenums]="linenums"></aio-code>
```

Comment (petebacondarwin) [inline aio/src/app/embedded/code/code-example.component.ts]: or possibly `ngClass`:

```html
<aio-code [ngClass]="{'headed-code':title, 'simple-code':!title}" [code]="code" [language]="language" [linenums]="linenums"></aio-code>
```

**Review by petebacondarwin — CHANGES_REQUESTED** [review]

**Comment by petebacondarwin** [comment]
Also these changes seem to have broken the unit tests, which you can check locally with `yarn test`

**Comment by sjtrimble** [comment]
@petebacondarwin thanks! I used ngClass like you recommended. I wasn't able to figure out yarn test (yarn test v0.21.3 | error Command "test" not found) so I may need some help there.

**Review by petebacondarwin — APPROVED** [review]

**Pull request merged**